src/utils/deep-merge.js

```diff
@@ -25,5 +35,5 @@ export default function deepMerge(
 }
 
 function isCloneable(obj: mixed) {
-  return Array.isArray(obj) || {}.toString.call(obj) == '[object Object]';
+  return typeof obj === 'object' || typeof obj === 'function';
```

Comment:
Why do you consider functions and `null` cloneable?

Reply:
i did exclude `null`, but function is an object. Lots of frameworks starting with jQuery use it as object with props, especially it's common if it's a function-constructor

Reply:
Could you point out where you exclude the `null`? How are you intended to `deepMerge` functions?

Reply:
Just pushed fresh code. Function merges the same way as any other object. We go through it's props and add them to target function, which is the prop itself of parent object